docs/serving/configuring-autoscaling.md

```diff
@@ -179,11 +181,39 @@ template.
 spec:
   template:
     metadata:
-      autoscaling.knative.dev/metric: concurrency
+      autoscaling.knative.dev/metric: cpu
       autoscaling.knative.dev/class: hpa.autoscaling.knative.dev
 ```
 
-## Additional resources
+# Implementing your own Pod Autoscaler
+
+The Pod Autoscaler custom resource allows you to implement your own autoscaler without changing anything else about the Knative Serving system.
+
+You can implement your own Pod Autoscaler if the requirements of your workload cannot be covered by the KPA or HPA, for example if you want to use a more specialized autoscaling algorithm or if you need to use a specialized set of metrics not supported by Knative out of the box.
+
+To implement your own Pod Autoscaler, you can create a reconciler that operates on your own class of Pod Autoscaler by copying a [Knative sample controller](https://github.com/knative/sample-controller) and modifying its configuration to suit your desired use case.
+
+For example, if your service's template YAML includes a class annotation like:
+```
+autoscaling.knative.dev/class: sample
+```
+Your reconciler would only want to reconcile PodAutoscaler resources with that target.
+
+The informer setup of your controller might look like this:
+
+```golang
+paInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+	FilterFunc: reconciler.AnnotationFilterFunc(autoscaling.ClassAnnotationKey, "sample", false),
+	Handler:    controller.HandleAll(impl.Enqueue),
+})
+```
+## Implementing required Knative autoscaling mechanisms
+
+* It is recommended to use the [`autoscaling-base-reconciler`](https://github.com/knative/serving/blob/master/pkg/reconciler/autoscaling/reconciler.go) as implemented in Knative Serving.
+* Ensure that you are calling `ReconcileSKS` from the `autoscaling-base-reconciler`.
+* If you want to use metrics collected by Knative like `concurrency`, ensure that you are using `ReconcileMetric` and `ReconcileMetricsService` to enable that system.
```

Comment:
Is this something different from https://knative.dev/docs/serving/installing-logging-metrics-traces/?

If it's that same, perhaps to align with terminology, we can say something like "If you have the observability plugins installed for collecting metrics,...."

Reply:
I'm not sure on this one - @markusthoemmes can you confirm?

Reply:
This is not related to each other. The docs here are referring to using these functions in a custom Go implementation, maybe we should make that clearer?

`ReconcileMetricsService` can vanish btw. It doesn't exist anymore.